src/record.rs

```diff
@@ -51,9 +51,8 @@ mod streaming_trait {
     impl<'t, T: Read + 't> Read for StreamingBody<'t, T> {
         fn read(&mut self, data: &mut [u8]) -> std::io::Result<usize> {
             let max_read = std::cmp::min(data.len(), *self.1 as usize);
-            self.0.read(&mut data[..max_read as usize]).map(|n| {
+            self.0.read(&mut data[..max_read]).inspect(|&n| {
                 *self.1 -= n as u64;
-                n
```

Comment:
So this was returning `n` but wasn't being assigned either. Am I reading that right?

Reply:
This is about clippy lint https://rust-lang.github.io/rust-clippy/master/index.html#manual_inspect.
The reason for this change was that `n` is only every *inspected* and never mutated or transformed. We're only mutating self while iterating. That's why `inspect` (which only gets a shared borrow) does the trick.